### PR TITLE
Delay or throw insertion when too many inactive parts

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add two settings to delay or throw error during insertion when there are too many inactive parts. This is useful when server fails to clean up parts quickly enough.

Detailed description / Documentation draft:

`inactive_parts_to_delay_insert`: If table contains at least that many inactive parts in single partition, artificially slow down insert into table.
`inactive_parts_to_throw_insert`: If more than this number inactive parts in single partition, throw 'Too many inactive parts ...' exception.

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -2369,6 +2369,26 @@ size_t MergeTreeData::getMaxPartsCountForPartition() const
         res = std::max(res, cur_count);
     }
 
+    if (inactive)
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeTreeData.cpp:2372]: The interface of this function is somewhat dirty.
Maybe simply add a separate function `getMaxInactivePartsCountForPartition`?

**Comment by alexey-milovidov** [comment]
> Some exception occurred :( 

`GithubException: 500 null`
This is GitHub downtime, https://www.githubstatus.com/history

**Comment by alexey-milovidov** [comment]
@amosbird What do you think about a slight change in the interface to make it more symmetric?

**Comment by amosbird** [comment]
> @amosbird What do you think about a slight change in the interface to make it more symmetric?

It's better. I'll do the refactoring.

**Pull request merged**